### ISSUE TITLE
训练的时候存在内存泄露

### ISSUE THREAD

**Issue by dayu1979** [by2101/OpenASR#12]
随着训练，内存一直升高，直到程序崩溃。大概2小时左右崩溃

**Comment by by2101** [comment]
应该不会吧，这个应该已经fix过了。原来是因为tr_loss变量少打了一个item()，所以导致变量一直拼接造成的。现在新版本应该没有了。如果定位到其他地方有问题的话麻烦提一下。

**Comment by dayu1979** [comment]
花了一天时间，找到问题了。
trainer.py这个文件中
tot_loss = tot_loss + batch_loss会造成内存泄露。具体原因不了解
改为:
tot_loss = tot_loss + batch_loss.cpu().detach().float()
就可以了。。。

**Comment by by2101** [comment]
我没找到batch_loss.float()的地方呀。这个地方应该不能detach，因为要传播梯度训练的。

**Comment by dayu1979** [comment]
说错了，这句:
tot_loss = tot_loss + batch_loss
你试下就知道了。这里在+到tot_loss的时候deteach，并没有影响batch_loss自身，因此梯度还能回传。如果不detach的话，
内存会一直。。。。
反正我32g内存，大概2小时崩溃。如果是内存特别大，可能这个问题不会暴露。

**Comment by by2101** [comment]
原来是这个地方。确实有可能有这个问题。这个地方实际上tot_loss是为了打log用的。batch_loss后面加个item转成float应该就可以了。
